File: music-domain/src/main/java/com/music/sale/domain/user/UserTerms.java
package com.music.sale.domain.user;

import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class UserTerms {
  private final Long id;
  private final Long userId;
  private final String title;
  private final String version;
  private final Boolean isAgreed;
  private final LocalDateTime createdAt;
  private final LocalDateTime updatedAt;

  public static UserTerms of(Long userId, String title, String version, Boolean isAgreed) {
    return UserTerms.builder()
        .userId(userId)
        .title(title)
        .version(version)
        .isAgreed(isAgreed)
        .build();
  }
}

File: music-domain/src/main/kotlin/com/music/sale/domain/user/UserTerms.kt
package com.music.sale.domain.user

data class UserTerms(
    val id: Long? = null,
    val userId: Long? = null,
    val title: String? = null,
    val version: String? = null,
    val isAgreed: Boolean? = null,
) {
    companion object {
        @JvmStatic
        fun of(
            userId: Long,
            title: String,
            version: String,
            isAgreed: Boolean,
        ): UserTerms =
            UserTerms(
                userId = userId,
                title = title,
                version = version,
                isAgreed = isAgreed,
            )
    }
}
